### PR TITLE
tg-upload-apis

### PR DESCRIPTION
**Pull request by nalwa**
This PR includes API for - **Api to upload bulk files on S3  CAN-234** , **API to delete created folder CAN-242**, **API to fetch list of uploaded files and folder CAN-238.**

This will require composer-update in order to load S3 packages.

@akuks  please review and merge the PR

### CODE REVIEW

File: app/Http/Controllers/UploadController.php [hunk 1]
```diff
@@ -3,12 +3,34 @@
 namespace App\Http\Controllers;
 
 use App\Http\Requests\AddFolderRequest;
+use Illuminate\Support\Facades\Storage;
```
Comment (akuks) [inline app/Http/Controllers/UploadController.php:6]: If not used, we can delete this

File: app/Http/Controllers/UploadController.php [hunk 2]
```diff
@@ -52,4 +73,234 @@ public function addFolder(AddFolderRequest $request)
...
+
+            $uploadFiles = [];
+            foreach($all['file'] as $k => $file){
+                $six_digit_random_number = random_int(100000, 999999);
+                $filename = User::ownerCode($user->id) . '_' . time() . '_' . $six_digit_random_number  .'.' . $file->getClientOriginalExtension(); 
+              
+                $s3Client = Aws::createS3Client();
+                    
+               $result = $s3Client->putObject([
+                    'Bucket' => 'canonizer-public-file',
```
Comment (akuks) [inline app/Http/Controllers/UploadController.php:128]: this should also comes from the env file

File: app/Http/Controllers/UploadController.php [hunk 3]
```diff
@@ -52,4 +73,234 @@ public function addFolder(AddFolderRequest $request)
...
+        } catch (\Throwable $e) {
+
+            return $this->resProvider->apiJsonResponse(400, trans('message.error.exception'), '', $e->getMessage());
+        }
+    }
+
+    
+ 
+    private function generateShortCode($strength = 9) {
+        $input = '0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ';
```
Comment (akuks) [inline app/Http/Controllers/UploadController.php:162]: Should be in config or env file

File: app/Http/Controllers/UploadController.php [hunk 4]
```diff
@@ -52,4 +73,234 @@ public function addFolder(AddFolderRequest $request)
...
+     */
+    public function folderDelete($id){
+        try{
+
+            $files = Upload::where('folder_id','=', $id)->get();
+
+            if(count($files) > 0){
+                $status = 400;
+                $message = trans('message.uploads.folder_has_files_can_not_delete');
+                return $this->resProvider->apiJsonResponse($status, $message, null, null);
```
Comment (akuks) [inline app/Http/Controllers/UploadController.php:247]: `return` statment should be moved out of if else

**Review by akuks — CHANGES_REQUESTED** [review]

**Comment by nalwa** [comment]
@akuks  change requested are fixed. Please review and merge PR.

**Pull request merged**